yjit/src/core.rs

```diff
@@ -996,20 +996,31 @@ impl Block {
         self.end_idx = end_idx;
     }
 
-    pub fn add_gc_obj_offset(self: &mut Block, ptr_offset: u32) {
-        self.gc_obj_offsets.push(ptr_offset);
-        incr_counter!(num_gc_obj_refs);
+    pub fn add_gc_obj_offsets(self: &mut Block, gc_offsets: Vec<u32>) {
+        for offset in gc_offsets {
+            self.gc_obj_offsets.push(offset);
+            incr_counter!(num_gc_obj_refs);
+        }
+        self.gc_obj_offsets.shrink_to_fit();
```

Comment:
Hmmm. This could actually make insertion an O(n) operation, no?

Reply:
`block.add_gc_obj_offset(offset)` was called N times, so this particular loop was O(n) from the beginning. But I assume you're talking about other functions as well.

A vector has `len` field and uses it to decide the reallocated size, instead of scanning N elements. Are you saying the reallocation cost is linear to the size of the buffer? Finding a space could be hard if the vector is too big, but I assume the cost doesn't necessarily grow when the vector is not too big.

Reply:
left my thoughts about this at https://github.com/ruby/ruby/pull/6739/files#r1025482671